### PR TITLE
[2.19.x] G-9102 Sort alerts by most recent

### PR DESCRIPTION
**Pull request by brhumphe**
#### What does this PR do?
Sorts alerts so the most recent alert is at the top of the alerts pane.

#### Select relevant component teams: 
@codice/ui 

#### What are the relevant tickets?
Fixes: #____

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Review by Lambeaux — APPROVED** [review]

**Review by bdeining — APPROVED** [review]

**Comment by bdeining** [comment]
build now

**Review by jordanwilking — APPROVED** [review]

**Comment by bdeining** [comment]
need to run the formatter

**Comment by brhumphe** [comment]
Just restarted the build after formatting

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/Alert.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ module.exports = Backbone.AssociatedModel.extend({
       this.set('id', Common.generateUUID())
     }
   },
-  getTimeComparator() {
+  getTime() {
```
Comment (mojogitoverhere) [inline ui/packages/catalog-ui-search/src/main/webapp/js/model/Alert.js:32]: ❗ Notifications are also created when users upload data from the UI. They rely on the same `getTimeComparator` method, so I recommend we revert this rename https://github.com/codice/ddf/blob/9c528c54f2540d7fb01e271fb83a00942ed20fea/ui/packages/catalog-ui-search/src/main/webapp/js/model/UploadBatch.js#L189-L191

**Review by mojogitoverhere — CHANGES_REQUESTED** [review]

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/Alert.js (hunk 1 above)
Comment (brhumphe) [reply ui/packages/catalog-ui-search/src/main/webapp/js/model/Alert.js:32]: They share a name but are different functions; the one in `UploadBatch.js` is comparing by `sentAt`, but this change compares by `when` so there shouldn't be any impact.

Besides, the name is totally erroneous: these do not return comparator functions, they only return values. I would prefer renaming the one in UploadBatch too;

Comment (mojogitoverhere) [reply ui/packages/catalog-ui-search/src/main/webapp/js/model/Alert.js:32]: Yes they use different values but the idea is the same - some date/time used for sorting purposes. I like that `Comparator` in the name implies that it is used somehow in sorting them, but I also agree that the name doesn't match what the method actually does so I'm on the fence. I'll leave it to your discretion

Comment (bdeining) [reply ui/packages/catalog-ui-search/src/main/webapp/js/model/Alert.js:32]: Screen Shot 2020-09-22 at 4 27 45 PM
this rename seems to be causing other issues during the hero

**Review by mojogitoverhere — APPROVED** [review]

**Review by brianfelix — APPROVED** [review]

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/Alert.js (hunk 1 above)
Comment (brhumphe) [reply ui/packages/catalog-ui-search/src/main/webapp/js/model/Alert.js]: Reverted the rename, kept only the call to `sort()`

**Comment by bdeining** [comment]
hero looks good

**Pull request merged**